server/jetstream_api.go

```diff
@@ -906,6 +924,8 @@ func (js *jetStream) apiDispatch(sub *subscription, c *client, acc *Account, sub
 			Domain:  js.config.Domain,
 			Dropped: int64(pending),
 		})
+	} else {
+		atomic.AddInt64(&js.apiInflight, 1)
```

Comment:
Don't we have the risk that - since the element has already been pushed and can be already be processed by now - that we at one point go negative? Sure, this one would bring it back to 0, but I would probably increment before the push and decrement if we have removed one with popOne(). That is, move the add after `retry:`, but decrement if popOne() is successful.

Reply:
Works for me, have updated.